File: src/main/kotlin/net/hackergarten/sessionapi/session/Session.java
package net.hackergarten.sessionapi.session;

import net.hackergarten.sessionapi.venue.Venue;

import javax.persistence.Entity;
import javax.persistence.GeneratedValue;
import javax.persistence.GenerationType;
import javax.persistence.Id;
import javax.persistence.ManyToOne;
import javax.persistence.Temporal;
import javax.persistence.TemporalType;
import java.util.Date;

@Entity
public class Session {

    private Long id;
    private String title;
    private Date date;
    private Venue venue;

    public Session() {
    }

    Session(String title, Date date, Venue venue) {
        this.title = title;
        this.date = date;
        this.venue = venue;
    }

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getTitle() {
        return title;
    }

    public void setTitle(String title) {
        this.title = title;
    }

    @Temporal(TemporalType.TIMESTAMP)
    public Date getDate() {
        return date;
    }

    public void setDate(Date date) {
        this.date = date;
    }

    @ManyToOne
    public Venue getVenue() {
        return venue;
    }

    public void setVenue(Venue venue) {
        this.venue = venue;
    }
}

File: src/main/kotlin/net/hackergarten/sessionapi/session/Session.kt
package net.hackergarten.sessionapi.session

import net.hackergarten.sessionapi.venue.Venue
import java.util.*
import javax.persistence.Entity
import javax.persistence.GeneratedValue
import javax.persistence.GenerationType
import javax.persistence.Id
import javax.persistence.ManyToOne
import javax.persistence.Temporal
import javax.persistence.TemporalType

@Entity
class Session() {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    var id: Long? = null
    var title: String = ""
    @Temporal(TemporalType.TIMESTAMP)
    var date: Date = Date(0)
    @ManyToOne
    var venue: Venue = Venue()

}
